mmv1/products/clouddeploy/DeployPolicy.yaml

```diff
@@ -0,0 +1,319 @@
+                  - "INVOKER_UNSPECIFIED"
+                  - "USER"
+                  - "DEPLOY_AUTOMATION"
+            - name: "actions"
+              type: Array
+              description: "Optional. Rollout actions to be restricted as part of the policy. If left empty, all actions will be restricted."
+              item_type:
+                type: Enum
+                enum_values:
+                  - "ROLLOUT_ACTIONS_UNSPECIFIED"
+                  - "ADVANCE"
+                  - "APPROVE"
+                  - "CANCEL"
+                  - "CREATE"
+                  - "IGNORE_JOB"
+                  - "RETRY_JOB"
+                  - "ROLLBACK"
+                  - "TERMINATE_JOBRUN"
+            - name: "timeWindows"
+              type: NestedObject
+              description: "Required. Time window within which actions are restricted."
+              properties:
+                - name: "timeZone"
+                  type: String
+                  description: "Required. The time zone in IANA format IANA Time Zone Database (e.g. America/New_York)."
+                  required: true
+                - name: oneTimeWindows
+                  type: Array
+                  description: "Optional. One-time windows within which actions are restricted."
+                  item_type:
+                    type: NestedObject
+                    properties:
+                      - name: "startDate"
+                        type: NestedObject
+                        description: "Required. Start date."
+                        required: true
+                        properties:
+                          - name: "year"
+                            type: Integer
+                            description: "Year of the date. Must be from 1 to 9999, or 0 to specify a date without a year."
+
+                          - name: month
+                            type: Integer
+                            description: "Month of a year. Must be from 1 to 12, or 0 to specify a year without a month and day."
+
+                          - name: day
+                            type: Integer
+                            description: "Day of a month. Must be from 1 to 31 and valid for the year and month, or 0 to specify a year by itself or a year and month where the day isn't significant."
+
+                      - name: "endDate"
+                        type: NestedObject
+                        description: "Required. End date."
+                        required: true
+                        properties:
+                          - name: "year"
+                            type: Integer
+                            description: "Year of the date. Must be from 1 to 9999, or 0 to specify a date without a year."
+
+                          - name: month
+                            type: Integer
+                            description: "Month of a year. Must be from 1 to 12, or 0 to specify a year without a month and day."
```

Comment:
Can you write a test that specifies 0 for this? 0 values tend to cause problems, so I'd like to make sure it's working correctly. Especially this case:
end_date set with month = 1
end_date updated to month = 0

Reply:
The description is generated from the shared Date message. In Deploy Policy, 0 is not allowed, so I adjusted the description.